src/com/magento/idea/magento2plugin/actions/generation/generator/MessageQueueClassGenerator.java

```diff
@@ -0,0 +1,145 @@
+     * @param messageQueueClassData MessageQueueHandlerData
+     * @param moduleName String
+     * @param project Project
+     */
+    public MessageQueueClassGenerator(
+            final MessageQueueClassData messageQueueClassData,
+            final String moduleName,
+            final Project project
+    ) {
+        super(project);
+
+        this.messageQueueClassDataName = messageQueueClassData;
+        this.directoryGenerator = DirectoryGenerator.getInstance();
+        this.fileFromTemplateGenerator = FileFromTemplateGenerator.getInstance(project);
+        this.validatorBundle = new ValidatorBundle();
+        this.commonBundle = new CommonBundle();
+        this.getFirstClassOfFile = GetFirstClassOfFile.getInstance();
+        this.project = project;
+        this.moduleName = moduleName;
+    }
+
+    @Override
+    public PsiFile generate(final String actionName) {
+        final PsiFile[] handlerFiles = new PsiFile[1];
+
+        WriteCommandAction.runWriteCommandAction(project, () -> {
+            PhpClass handler = GetPhpClassByFQN.getInstance(project).execute(
+                    messageQueueClassDataName.getFqn()
+            );
+
+            if (handler != null) {
+                final String errorMessage = this.validatorBundle.message(
+                        "validator.file.alreadyExists",
+                        "Handler Class"
+                );
+                JOptionPane.showMessageDialog(
+                        null,
+                        errorMessage,
+                        commonBundle.message("common.error"),
+                        JOptionPane.ERROR_MESSAGE
+                );
+
+                return;
+            }
+
+            handler = createHandlerClass(actionName);
+
+            if (handler == null) {
+                final String errorMessage = this.validatorBundle.message(
+                        "validator.file.cantBeCreated",
+                        "Handler Class"
+                );
+                JOptionPane.showMessageDialog(
+                        null,
+                        errorMessage,
+                        commonBundle.message("common.error"),
+                        JOptionPane.ERROR_MESSAGE
+                );
+
+                return;
+            }
+
+            handlerFiles[0] = handler.getContainingFile();
+        });
+
+        return handlerFiles[0];
+    }
+
+    @Override
+    protected void fillAttributes(final Properties attributes) {
+        attributes.setProperty("NAMESPACE", messageQueueClassDataName.getNamespace());
+        attributes.setProperty("CLASS_NAME", messageQueueClassDataName.getName());
+    }
+
+    private PhpClass createHandlerClass(final String actionName) {
+        PsiDirectory parentDirectory = ModuleIndex.getInstance(project)
+                .getModuleDirectoryByModuleName(this.moduleName);
+        final PsiFile handlerFile;
+        final String[] handlerDirectories = messageQueueClassDataName.getPath().split(
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [RE: File.separator used for regular expression (RE_CANT_USE_FILE_SEPARATOR_AS_REGULAR_EXPRESSION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#re-file-separator-used-for-regular-expression-re-cant-use-file-separator-as-regular-expression).
The code here uses File.separator where a regular expression is required. This will fail on Windows platforms, where the File.separator is a backslash, which is interpreted in a regular expression as an escape character. Among other options, you can just use File.separatorChar=='\\' ? "\\\\" : File.separator instead of File.separator

Reply:
This won’t fail, because we use our own implementation of the `File` class. See imports.